### ISSUE TITLE
Add temporal recurrence and culprit stability to rank-straggler evidence

### ISSUE THREAD

**Issue by abhinavsriva** [traceopt-ai/traceml#374]
## Problem

TraceML currently decides rank-straggler evidence from per-rank averages over an aligned window. The same window-level finding can therefore describe very different behavior:

- one rank repeatedly exhibits the visible skew;
- the implicated rank rotates across steps; or
- a small number of transient steps dominate the average.

The canonical Step Time window already retains selected-clock facts for every aligned step and global rank, but the current straggler builder reads only `facts.average`:

- [`StepTimeRankFacts.steps`](https://github.com/traceopt-ai/traceml/blob/c91a0142f9f7a8a3d578e6c511696d44c97be4ad/src/traceml_ai/step_time/model.py#L151-L165)
- [current window-average straggler evidence](https://github.com/traceopt-ai/traceml/blob/c91a0142f9f7a8a3d578e6c511696d44c97be4ad/src/traceml_ai/diagnostics/step_time/context.py#L193-L235)

## Goal

Add temporal recurrence and culprit-stability as **supporting evidence** for an existing rank-straggler finding. Do not introduce a new root-cause model or change the current verdict/severity in this issue.

For each eligible aligned step:

1. Reuse the window's selected clock, eligible-rank cohort, training-strategy semantics, and current culprit/victim rule.
2. For DDP, use the same visible backward metric; for FSDP, use the same forward-plus-backward metric.
3. Compute the same normalized visible-gap score used by the current policy.
4. Count a step as affected when it reaches the existing warning threshold.

Publish structured evidence with unambiguous numerators and denominators:

- `eligible_steps`
- `affected_steps`
- `persistence_fraction = affected_steps / eligible_steps`
- `dominant_culprit_rank`
- `dominant_culprit_steps`
- `culprit_stability_fraction = dominant_culprit_steps / affected_steps`
- median and p90 per-step visible-gap score for affected steps

Keep recurrence separate from culprit stability: a job can exhibit recurring skew while the implicated rank changes.

## User-facing behavior

When an existing rank-straggler finding is present and coverage is sufficient, add one concise evidence line, for example:

```text
Observed rank skew on 84/100 eligible steps; r2 was implicated on 80/84 affected steps.
```

Below the existing confident-diagnosis window size, report the observed counts as limited evidence; do not label the behavior "persistent." Missing or ineligible steps must reduce reported coverage, never become zeros.

## Semantics and guardrails

- Call this observed rank-skew recurrence, not counterfactual slowdown or causal attribution.
- This does **not** reproduce the paper's per-step slowdown: TraceML has no simulated `T_ideal`.
- Do not use naive maximum total-step time to choose a culprit; distributed synchronization can equalize or invert visible phase durations.
- Do not change diagnosis status, cause, score, or severity in the first implementation.
- If rank-local export windows from #229/#231 no longer preserve individual steps, mark recurrence unavailable unless equivalent counts were computed before reduction. Never treat one export window as one training step.

## Acceptance criteria

- [ ] A shared core fact model computes recurrence from `StepTimeWindow.rank_facts`, not presentation-layer series.
- [ ] Fixed-rank, rotating-rank, isolated-transient, and balanced synthetic windows produce deterministic counts and culprit ties.
- [ ] DDP and FSDP use their existing direction-aware visible metrics and pair selection.
- [ ] Partial-rank, missing-metric, selected-clock, low-step-count, and windowed-export cases expose honest coverage or abstain.
- [ ] Structured evidence reaches live diagnostics and `final_summary.json` through the canonical Step Time projection.
- [ ] Terminal/dashboard/report wording uses the same facts and adds no competing verdict.
- [ ] Existing aggregate straggler scores and golden outputs remain unchanged except for the new supporting evidence.

## Relationship to existing work

- Complements #229/#231; it does not replace their telemetry-volume design.
- Does not fix the backward-origin blind spot documented in #252; explicit collective timing remains #224.
- Is unrelated to the generic trend-display bug in #356.

## Research reference

[Lin et al., *Understanding Stragglers in Large Model Training Using What-if Analysis*, OSDI '25](https://www.usenix.org/system/files/osdi25-lin-jinkun.pdf), Section 4.2 and Figure 4, motivates separating recurring slowdown from isolated steps. That result used dependency-aware what-if simulation on ByteDance traces; it is inspiration for this observational TraceML evidence, not a directly reusable metric.
